job/jobcache.go

```diff
@@ -1,36 +1,86 @@
 package job
 
 import (
+	"os"
+	"os/signal"
 	"sync"
+	"time"
 )
 
-var (
-	AllJobs = &JobCache{
-		Jobs:   make(map[string]*Job),
-		rwLock: sync.Mutex{},
+type JobCache interface {
+	Get(id string) *Job
+	GetAll() map[string]*Job
+	Set(j *Job)
+	Delete(id string)
+	Persist() error
+}
+
+type MemoryJobCache struct {
+	Jobs            map[string]*Job
+	rwLock          sync.Mutex
+	jobDB           JobDB
+	persistWaitTime time.Duration
+}
+
+func NewMemoryJobCache(jobDB JobDB, persistWaitTime time.Duration) *MemoryJobCache {
+	if persistWaitTime == time.Duration(0) {
```

Comment:
@ajvb Go will let you just do `if persistWaitTime == 0`, since `time.Duration` is defined as `type Duration int64`; the literal `0` will be interpreted as the number type necessary for that comparison to be valid, since `persistWaitTime` is a `time.Duration` is an `int64`.